### ISSUE TITLE
Feedback for Development Release 4

### ISSUE THREAD

**Issue by wildex999** [wildex999/TickDynamic#26]
I have just released Development version 4, and I'm getting close to the final 0.2.0 release, before I start working on the next feature set.

I would appreciate it if you could take the time to test a release for your server(Taking world backup first!!!) and giving some feedback on what works and what doesn't while running on a loaded server.

I have released dev4 for KCauldron, Forge 1.7.10 and Forge 1.8.9. You can find them under releases here: https://github.com/wildex999/TickDynamic/releases

**Changes for dev4**
- Implemented a missing method which could have causes problems. (Fixes player respawn in KCauldron)
- Handle case where removing a Entity/Tile which is the last one iterating, causes it to fail. (Should stop desync message for players in KCauldron)
- Better debugging messages.

**Changes for dev3**
- Worlds should now properly unload without leaking
- Added new command: 'reloadgroups', which will reload all groups, and reassign any loaded Entities or TileEntities into the new group setup. Note: This is currently a brute force approach, I would appreciate any feedback on the impact this command has on a server under load.
- Config fixes, it should now more properly respect global groups and default settings.
- Entities should now properly unload
- Other stability fixes

**Known Problems**
- None yet

**Reporting problem**
- If you have a problem you can reproduce, please set 'debug = true' in the config and upload a log when reporting the problem.

**Comment by spannerman79** [comment]
I'll test the Forge 1.7.10 build soon(ish).  Long story short I had to stop using kcauldron and move to a vanilla forge instance.  It wasn't by choice (many of my users have some sort of development issue being Autistic, ADHD, etc).

**Comment by wildex999** [comment]
@spannerman79 Yeah, so far KCauldron has been more problem than it's worth, so I don't think I'll support it past 0.2. For 1.8 Sponge seems to be the thing =)

**Comment by spannerman79** [comment]
> For 1.8 Sponge seems to be the thing =)

Sure is :+1: Not there yet with the number of mods or their function to move fully over to SpongeForge .._yet_

**Comment by filthy1337** [comment]
I have tested TickDynamic Dev v3 on my Feed the Beast Infinity Evolved server (2.3.5) with KCauldron-1.7.10-1614.193. There is no crash, but the server just kinda stops before it gets fully up on-line. :/
Is there anything I could try to do to make this mod work for me?

**Comment by spannerman79** [comment]
@filthy1337 Logs? Use gist or upload the uncompressed log here

**Comment by badgernz** [comment]
I have run it on a loaded Infinity server aswell but with KCauldron:1.7.10-1614.174 and have had no loading errors, just crashes when you try use tick profiler and/or opis 

http://pastebin.com/AKXCs91M

**Comment by filthy1337** [comment]
https://gist.githubusercontent.com/filthy1337/a4343c62b08ba5ab8714/raw/297a2553c189b3bad0808d4da66e94ad8041d19f/gistfile1.txt

**Comment by spannerman79** [comment]
> just crashes when you try use tick profiler and/or opis

@badgernz Use mapless Opis from [here](https://gitlab.prok.pw/KCauldron/KCauldron/wikis/patched-mods) or [here](https://github.com/Gamealition/Opis/releases)

**Comment by wildex999** [comment]
@filthy1337 Could you upload your logs somehwere(pastebin, gist etc.) and post it here? "kinda stops" doesn't help me much unfortunately =(

@badgernz Yeah, compatibility with other mods which heavily modifies/depends on the world loop(Like Opis and Tick Profiler) will work poorly with TickDynamic.
I might make it compatible on Forge servers at some point, but KCauldron is just too much of a minefield to even try.

Thank you both for taking the time to test and report here =)

**Comment by filthy1337** [comment]
Hey! I reinstalled my server (FTB Infinity Evolved 2.3.5) from scratch and and now TickDynamic Dev v3 works as it should. I guess the problem was not in TickDynamic itself (although the problem appeared only after adding TickDynamic and went away by disabling it?), but some other thing that I just couldn't figure out... :/ Well, anyway thanks for this great mod!! Keep up the good work!!

**Comment by wildex999** [comment]
@filthy1337 Reinstalled as in started a new world?
There might be certain things that doesn't cause problems unless they exist in the world etc.
If it suddenly starts happening again, it would be nice to know what caused it =)

**Comment by filthy1337** [comment]
I actually installed the whole server again from the ground up. :) Downloaded the FTB pack, KCauldron and TickDynamic just so that there was no possibility of unrelated errors confusing me even more! :D If it stops working again I now have a better chance to find out what could have caused the problem in the first place. I'll keep you posted...

**Comment by spannerman79** [comment]
Ok so I want to test out the forge 1.7.10 version of the dev release ( https://github.com/wildex999/TickDynamic/releases/tag/Forge1.7.10_0.2.0-dev3 ), but to really test it out i want to be able to add some groups in.

Does anyone have any groups in their config file setup that they are willing to share?

**Comment by spannerman79** [comment]
@wildex99 do you have some recommended group settings in order for me to test out your groups option on my forge 1.7.10 server?

**Comment by wildex999** [comment]
@spannerman79 Not really. I originally added the group system to allow for excluding mods which was not compatible with Dynamic Ticking.
When I test I mostly just throw Entities like Pigmen into their own groups just to see that it works.

I would recommend you just start putting things you think use a lot of time(Enemy mobs for example) into their own group, and then check how much time they actually use to further plan out the groups.
The group reload should work now, so you can do rapid testing.

As I might have mentioned earlier, I don't yet have a full server to actually test the mod on, which is why I created this thread =)

**Comment by wildex999** [comment]
I have now released Development version 4. Feedback continues in this ticket for now.

This update is mostly targeted at KCauldron users, but the problems might appear in other servers also, so upgrade if you have any problems.

**Comment by filthy1337** [comment]
Tested this dev build 4 and got a crash. [crash-2016-02-20_14.59.27-server.txt](https://github.com/wildex999/TickDynamic/files/139125/crash-2016-02-20_14.59.27-server.txt)

**Comment by wildex999** [comment]
Hey @filthy1337 
Could you give this one a try?
https://github.com/wildex999/TickDynamic/releases/tag/KCauldron_0.2.0-dev5pre

**Comment by filthy1337** [comment]
I will definitely test it! Thanks! :)

**Comment by filthy1337** [comment]
For twice now I have had a client crash which has never happened before... Is it a coincidence or TickDynamic related?
[crash-2016-02-21_01.40.05-client.txt](https://github.com/wildex999/TickDynamic/files/139565/crash-2016-02-21_01.40.05-client.txt)
[crash-2016-02-21_12.27.57-client.txt](https://github.com/wildex999/TickDynamic/files/139566/crash-2016-02-21_12.27.57-client.txt)

**Comment by spannerman79** [comment]
@filthy1337 open an issue over at the correct issue tracker - http://minecraft.curseforge.com/projects/decocraft2/issues

**Comment by wildex999** [comment]
Unless you are running TickDynamic on the client, I can't see how it can cause this. Won't rule it out entirely though. TickDynamic has nothing to do with the chunks being sent to the client, and this seems to be entirely a client error. 

The crashing mod needs to throw in a "instanceof" check before casting it seems to me ;)

**Comment by filthy1337** [comment]
Okay, I got a new error now:
[crash-2016-02-26_23.19.59-server.txt](https://github.com/wildex999/TickDynamic/files/149272/crash-2016-02-26_23.19.59-server.txt)

**Comment by enterprise12nx** [comment]
I moved to thermos from Kcauldron when it started running smoother, and then i saw Kcauldron dissapeared, i remember @spannerman79 
Hoping some support can come to Thermos, but i can understand working towards sponge (hurry up!)
@wildex999
